Content.Server/Xenoarchaeology/Artifact/XAE/XAEChargeBatterySystem.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using Content.Shared.Xenoarchaeology.Artifact;
+using Content.Shared.Xenoarchaeology.Artifact.XAE;
+using Robust.Server.GameObjects;
+using XAEChargeBatteryComponent = Content.Server.Xenoarchaeology.Artifact.XAE.Components.XAEChargeBatteryComponent;
+
+namespace Content.Server.Xenoarchaeology.Artifact.XAE;
+
+public sealed class XAEChargeBatterySystem : BaseXAESystem<XAEChargeBatteryComponent>
+{
+    [Dependency] private readonly BatterySystem _battery = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly TransformSystem _transform = default!;
+
+    /// <inheritdoc />
+    protected override void OnActivated(Entity<XAEChargeBatteryComponent> ent, ref XenoArtifactNodeActivatedEvent args)
+    {
+        var chargeBatteryComponent = ent.Comp;
+
+        var artifactCoordinates = _transform.GetMapCoordinates(ent);
```

Comment:
you end up getting the coords a lot so it's probably better if you just pass it through the event to begin with.

Reply:
Funny thing - you already did that. i forgot to use coords from args.